ADO vscode pipeline fails: esbuild platform binary dropped by 'npm ci --no-optional'

### Summary

The `azure-dev - vscode` ADO pipeline (`BuildAndTest` → `BuildVscode {Linux,Mac,Windows}` → **Test** task) fails on **all three OSes** for any PR that touches `ext/vscode`. The failure is a build/test tooling issue, not a product code issue.

### Symptom

All three OS Test tasks fail with the platform-specific esbuild binary missing:

```
Error: The package "@esbuild/linux-x64"  could not be found, and is needed by esbuild.   (Linux)
Error: The package "@esbuild/darwin-x64" could not be found, and is needed by esbuild.    (Mac)
Error: The package "@esbuild/win32-x64"  could not be found, and is needed by esbuild.    (Windows)
##[error]PowerShell exited with code '1'.
```

esbuild's error even spells out the cause:

> If you are installing esbuild with npm, make sure that you don't specify the `--no-optional` or `--omit=optional` flags. The `optionalDependencies` feature of `package.json` is used by esbuild to install the correct binary executable for your current platform.

(On Windows there is also an intermittent VS Code `1.128.1` download **checksum mismatch**, but the esbuild error is the common, deterministic root cause across all three OSes.)

### Root cause

The ADO job installs dependencies with **`npm ci --no-optional`**:

- `eng/pipelines/templates/jobs/vscode-build.yml` (Install dependencies step)

esbuild ships its platform binary (`@esbuild/<platform>`) as an **optional dependency**, so `--no-optional` omits it. Both `npm run build:esbuild` and the extension-host test runner (which bundles test files via esbuild) then fail at runtime.

### Why it started failing now

- The ADO job runs `npm i -g npm` (upgrades to the latest npm) **before** `npm ci --no-optional`. Current npm strictly honors `--no-optional` and drops esbuild's optional platform packages.
- GitHub Actions `vscode-ci.yml` (`build-test`) runs the **same** `./ci-test.ps1` with `npm ci --no-optional` but uses `setup-node`'s bundled npm, which still resolves the platform binary — so GitHub is **green on all 3 OSes** while ADO is red. This divergence is why the two CIs disagree.
- Because both `npm i -g npm` and the VS Code "stable" download float, this regressed silently; older `ext/vscode` PRs passed, but it now blocks every current `ext/vscode` PR.

### Proposed fix

Remove `--no-optional` from the dependency install step so esbuild's platform binary is installed:

```diff
-          npm ci --no-optional
+          npm ci
```

in `eng/pipelines/templates/jobs/vscode-build.yml` (and mirror in `.github/workflows/vscode-ci.yml` for parity). Alternatives: pin the global npm version, or add the current-platform esbuild package explicitly.

### Evidence

- Failing ADO build: `azure-dev - vscode` buildId `6560427` (also reproduced in `6560377`), org `azure-sdk`, project `internal`.
- Passing GitHub run: `vscode-ci` `build-test` (ubuntu/macos/windows) on the same commit.
- Discovered while working on #9130 (PR #9136), whose only diff is 4 files under `ext/vscode/src` + `CHANGELOG.md` — none touch build/test tooling.

--- a/.github/workflows/vscode-ci.yml
+++ b/.github/workflows/vscode-ci.yml
@@ -63,7 +63,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          npm ci --no-optional
+          npm ci
         working-directory: ./ext/vscode
 
       - name: Test

--- a/eng/pipelines/templates/jobs/vscode-build.yml
+++ b/eng/pipelines/templates/jobs/vscode-build.yml
@@ -35,7 +35,7 @@ jobs:
 
       - pwsh: |
           npm i -g npm @vscode/vsce
-          npm ci --no-optional
+          npm ci
         workingDirectory: ext/vscode
         displayName: Install dependencies
 

